src/ImageStats/Histogram.h

```diff
@@ -14,29 +14,45 @@
 
 namespace carta {
 
-struct HistogramResults {
-    int num_bins;
-    float bin_width;
-    float bin_center;
-    std::vector<int> histogram_bins;
-};
-
 class Histogram {
-    int _num_bins;
-    float _bin_width;
-    float _min_val;
-    std::vector<int> _hist;
-    const std::vector<float>& _data;
+    float _min_val;                   // lower bound of the histogram (inclusive)
+    float _max_val;                   // upper bound of the histogram (inclusive)
+    int _num_bins;                    // number of bins
+    float _bin_width;                 // bin width
+    float _bin_center;                // bin center
+    std::vector<int> _histogram_bins; // histogram bin counts
+
+    void Fill(const std::vector<float>&);
+    static bool ConsistencyCheck(const Histogram&, const Histogram&);
 
 public:
+    Histogram() = default; // required to create empty histograms used in references
     Histogram(int num_bins, float min_value, float max_value, const std::vector<float>& data);
-    Histogram(Histogram& h, tbb::split);
-
-    void operator()(const tbb::blocked_range<size_t>& r);
-    void join(Histogram& h); // NOLINT
-    void setup_bins();
 
-    HistogramResults GetHistogram() const;
+    Histogram(const Histogram& h);
+
+    bool join(const Histogram& h); // NOLINT
+
+    float GetMinVal() const {
```

Comment:
I'm not 100% convinced that we need to use setters and getters for these, rather than accessing the member attributes (as we did with the results struct), but I don't feel that strongly about it. Python instincts. :)

Reply:
I think getters (with no setters) are the right approach here, because creating a histogram, and then simply changing the min or max values using `histogram.min_val = 20` will make the results incorrect.

Reply:
Fair enough; it's safer to prevent this.

Reply:
I didn't add setters for that reason, and we don't want to corrupt the histogram behavior. I don't want to use direct access to member attributes because that makes it a bit harder to do changes in the class behavior without having an impact on the rest of the code, see for example, this PR :)